compiler/optimizer/DebuggingCounters.cpp

```diff
@@ -234,7 +234,7 @@ void TR_DebuggingCounters::report()
          int32_t deviation = ((counterInfo->delta + 1)*counterInfo->bucketSize);
 
          if (deviation != INT_MAX)
-            fprintf(output, "Name: [%31s (%5d)] dynamic : (%5.2lf ) static : (%5.2lf ) [%lu]\n",
+            fprintf(output, "Name: [%31s (%5d)] dynamic : (%5.2lf ) static : (%5.2lf ) [%llu]\n",
```

Comment:
For portability, it would be better to use the `#defines` from `omrformatconsts.h` here and in similar places:

```suggestion
            fprintf(output, "Name: [%31s (%5d)] dynamic : (%5.2lf ) static : (%5.2lf ) [%" OMR_PRIu64 "]\n",
```

Reply:
I agree that would be better, though that change should really be applied more broadly and consistently.

@Leonardo2718 would you be willing to address this topic via a follow-on beginner issue?

Reply:
Let me clarify my request, because the original could have been interpreted as me asking @Leonardo2718 to do the work. What I meant to ask was: would you be willing for this topic to be addressed via a follow-on beginner issue (i.e. not fixed via this PR but made into an issue suitable for a beginner to pick up) ?

Reply:
As an aside would we get a compiler warning on some platforms if we did _not_ use `OMR_PRIu64`? I have the pipe dream of enabling warnings as errors piecemeal in the future.

Reply:
I share that pipe dream, which is one reason why I periodically squash a few of these warnings as an attempt to get closer to that reality.

Reply:
> As an aside would we get a compiler warning on some platforms if we did _not_ use OMR_PRIu64?

Yes, this is a risk. I have seen cases in the past where removing this kind of warning on one platform introduced a similar warning on another platform. That's the reason I generally recommend using the `OMR_PRI*` macros. I haven't inspected the code here closely enough to tell whether a warning will actually be introduced on another platform, though.

We should definitely have an issue to track making use of the `OMR_PRI*` macros more broadly (I thought we already did but we apparently don't).

Reply:
Just to tie up this conversation, the underlying line of code commented on here has, in the interim, been updated and this pull request preserves the use of OMR_PRI* macros.